### PR TITLE
Add UI to clone lessons to another script

### PR DESCRIPTION
**Pull request by bethanyaconnor**
Finishes [PLAT-963](https://codedotorg.atlassian.net/browse/PLAT-963). Adds a button to the lesson tokens to clone the level to another script.

Successful clone:

https://user-images.githubusercontent.com/46464143/118315602-93ad9a00-b4aa-11eb-9492-203231b7873a.mov

Clone with errors:

https://user-images.githubusercontent.com/46464143/118315792-cd7ea080-b4aa-11eb-83ee-8e8f3c50b4ef.mov

We had talked about adding script name autocomplete. I didn't add that right now as they would've complicated this already sizeable PR, it's usable as is, and there is a decent error message if the user misspells the script name. Definitely possible as a future improvement if people think it's worth it.

## Testing story

lot of manual testing, unit tests

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Comment by tess323** [comment]
LGTM - agreed we can wait for feedback on the autocomplete.  Thank you for all of your hard work on this!

File: apps/src/lib/levelbuilder/script-editor/CloneLessonDialog.jsx [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+  onCloneClick = () => {
+    this.setState({saving: true});
+    const csrfContainer = document.querySelector('meta[name="csrf-token"]');
+    let success = false;
+
+    return fetch(`/lessons/${this.props.lessonId}/clone`, {
+      method: 'POST',
+      body: JSON.stringify({
+        destinationScriptName: this.state.destinationScript
```
Comment (dmcavoy) [inline apps/src/lib/levelbuilder/script-editor/CloneLessonDialog.jsx]: I have renaming on the brain and it reminded me that we are trying to move from Script to Unit. Should we try to use Unit here?

**Comment by dmcavoy** [comment]
I pulled this locally to test how it worked to move things from aiml-2021 to csd7-2021. I ran into an issue when a unit has no lesson groups.

This line in `  def self.copy_to_script(original_lesson, destination_script)`

```
    copied_lesson.lesson_group_id = destination_script.lesson_groups.last.id
```

Ends up not finding a lesson group if the script hasn't had one added to it. It might be worth adding something to create a lesson group that is not user-facing if one does not exist? Or giving a warning that lets the user know they should go make one. The error I got was `undefined method `id' for nil:NilClass`

**Comment by dmcavoy** [comment]
Also looks like you can't go to a unit and save a lesson group with nothing in it.  https://codedotorg.atlassian.net/browse/PLAT-1065

Worked great once I added a lesson so it forced it to have a lesson group

**Comment by dmcavoy** [comment]
Love how simple the UI for this ending up being!

File: apps/src/lib/levelbuilder/script-editor/CloneLessonDialog.jsx (hunk 1 above)
Comment (bethanyaconnor) [reply apps/src/lib/levelbuilder/script-editor/CloneLessonDialog.jsx]: Sounds good! Is the script to unit change a known move? Wondering if/how I should update the text on this dialog -- maybe something like "Which script/unit do you want to clone this lesson to?"

**Comment by bethanyaconnor** [comment]
> Ends up not finding a lesson group if the script hasn't had one added to it. It might be worth adding something to create a lesson group that is not user-facing if one does not exist?

oh good call. I noticed this when adding unit tests but that was before the empty script discussion. Should be easy to add a level group if one doesn't exist. If it's not, I'll do it in a follow-up PR

File: apps/src/lib/levelbuilder/script-editor/CloneLessonDialog.jsx (hunk 1 above)
Comment (dmcavoy) [reply apps/src/lib/levelbuilder/script-editor/CloneLessonDialog.jsx]: It's part of our Q2 plan - https://codedotorg.atlassian.net/browse/PLAT-791. I don't know how deep of a dive we were planning to try to do to move away from script

**Comment by bethanyaconnor** [comment]
> Should be easy to add a level group if one doesn't exist. If it's not, I'll do it in a follow-up PR

I managed to break seeding on my local environment trying to test adding a lesson group, so I'm just going to throw an error and then have a better fix in a follow-up

**Review by dmcavoy — APPROVED** [review]
Nice improvements! Excited to see this go live!

**Pull request merged**